### ISSUE TITLE
MVS Mode game sound issues

### ISSUE THREAD

**Issue by OBurgerGuy** [libretro/fbalpha#163]
.42 has is missing sound in parts of these games in mvs mode:
nam1975
bstars
bstarsh
Possibly other clones of same games as well.

**Comment by OBurgerGuy** [comment]
sound works fine in aes mode

**Comment by barbudreadmon** [comment]
Could you be a little more precise ? I tried those games and didn't notice anything.

**Comment by OBurgerGuy** [comment]
run them in retropie under usa bios or latest unibios and theres no sound in nam 1975 durning Muckly text or in game, baseball starts no sound at all and fighters history dynamite no sound in selection screen or in game works fine with lr-fba2012 tho just not in current .42

**Comment by barbudreadmon** [comment]
Next time, don't say "mvs mode", making false report doesn't help with tracking an issue...

The issue with sound only happens if you select a mvs usa bios (which is not the default mvs bios) or if you select usa+arcade in unibios (which is not the default setting for unibios).

I'll check if it happens with standalone version, if it does i'll report the bug upstream.

And please don't say "run them in retropie", i don't have retropie, you don't need retropie to run this core. Saying it that way makes your issue sounds like a retropie specific one.

**Comment by OBurgerGuy** [comment]
my apologies, i dont know your etiquette, and you did say "Could you be a little more precise ? I tried those games and didn't notice anything." next time you shoule be more specific in your request...

also i used the terminologies i am familiar with as i do not know your "INDUSTRY" terms so from my point of view "mvs mode" is a thing and as for "default bios settings" again you didn't specify anything, as for retropie... this is the git for lr-fba which is most popularly used in a retropie environment, i did assume that the developers would have access to and use the most used platform for their software to develop and test for it, and running on retropie was just how i did it and got these results, as i did not know if it was retropie or core specific, maybe try not to sound and/or be so elitist about your reply when someone actually finds a problem and takes their time to report it to help the community.

thank you for your attention to this matter.

**Comment by barbudreadmon** [comment]
So what ? Do you think i have plenty of time to search for the real issue when someone reports the wrong issue ? Are you telling me you didn't notice the sound worked fine on mvs mode before you changed country to usa twice (with mvs and with unibios) ? Pointing that out is elitist ?

I don't know what's the most popular environment using this core, there are quite a few retrogaming distro for raspberry, recalbox being probably as popular as retropie, and there are tons of other devices able to run this. Thinking i should own a retropie setup because **you think** it's the most used platform is quite messed up. So if suddenly this core becomes really popular with ios users (perhaps they are already the largest community using this core, i don't know), i should buy an iphone X ?

Thanks for the report, just be more thorough with what you did the next time, so i won't search in the wrong direction.

**Comment by barbudreadmon** [comment]
Also, the fix seems to include disabling a speedhack in the neogeo code. I'll probably work on some dipswitch to enable/disable this speedhack when i have some time, until then don't use usa bioses.

**Comment by OBurgerGuy** [comment]
no i don't assume you have time for frivolities, but i do assume you take this project seriously and want people to report bugs, otherwise you are doing all the debugging all your self, kike the days before places like this... circa 1997! and no i never noticed the sound working, BECAUSE IT DIDN'T, it happens just like i said, would you like an effing video? i only changed it to usa region when it didnt work under default region, NO your attitude and assumptions i was describing as elitest i.e. "Next time, don't say "mvs mode", making false report doesn't help with tracking an issue..." and just to let you know it didn't work for me under any bios region, stock or unibios, except forcing aes, no sound still. and i never posted a fake issue, this is a real issue, i didnt post here until i tried EVERYTHING, EVERY combination of bioses, regions and such, because i needed a fix NOW, the last thing i wanted to do is get into a pissing match with a mod or coder and wait with my hand on may ass for a fix and release, so don't assume i didn't try everything before posting and only when i was absolutely out of answers, i figured this is the place to go, you go to the guys who make it, you are really angry, its not my fault, on top of which i'm not some chump just starting out in emulation, and I PAID for my unibios files from Razoola, i do not use the free version, and in case you ask yes, the paid and free unibios has the same issue, the only way the mentioned roms run fine is under 2012 version. and what the hell does an iphone x have to do with anything, if you are making a popularity comparison i would say android devices are the most popular by sheer sales numbers, but only because they are usually cheaper and poor morons buy them every six months... ::cough:: samsung.

again thank you for your attention in this matter and in the future please try to be more respectful of others as you do not know me and have no right treating me with the disrespectful attitude that you have been portraying, as i have not done this to you.

**Comment by barbudreadmon** [comment]
Ok, this is starting to make sense now, so basically you are saying you messed with your neogeo.zip to include your bios, could you post the file list with md5/crc in your neogeo.zip ? 

The only possible reason you would have an issue with "any" mvs bios would be that asia-s3.rom, sp-s2.sp1 and sp-s.sp1 are missing from your neogeo.zip. If you think i was rude before (you report an issue that doesn't exist because there is no issue with any mvs bios besides the usa ones, and you insist, that kinda annoys me), prepare yourself if you posted an issue because of incomplete romset.

**Comment by OBurgerGuy** [comment]
![no sound bios](https://user-images.githubusercontent.com/27814499/33544135-29e74e9e-d89f-11e7-9d73-99ff2ca79685.jpg)
![working sound bios](https://user-images.githubusercontent.com/27814499/33544137-2b6ad88a-d89f-11e7-97ab-3921f664d7fd.jpg)

**Comment by OBurgerGuy** [comment]
Video of problem: https://ufile.io/3vaxv

**Comment by barbudreadmon** [comment]
For the remainder, this is what is written in the "homepage" of this repository : 

> Use clrmamepro (which run fine on linux with wine and macos with crossover) to build valid romsets with dats from the dats directory. fba dats are NOT mame dats. Don't report issues if you didn't build a valid romset.

So, half of the normal content of neogeo.zip for fba is missing in your screenshot, which means you didn't respect that rule. How about following rules and completing your neogeo.zip file ?

Second thing, how am i supposed to confirm your unibios is not configured with usa country (which, as i have been saying since my second post, is what trigger your issue, so you shouldn't do that) with your video ?

**Comment by OBurgerGuy** [comment]
ok, i dont understand, as these THREE games ran fine before in fba2012, before you guys changed over and started a new version/branch? and with the bios zip i showed you, and they still do... Razoola worked with you to get crc checking off roms or bios or whatever for this new version, and the other thing is what files am i missing? is there a list of required files to be included in neogeo.zip? give me a file list for the bios zip, i will complete it and continue investigating this issue. also the country setting has no bearing in retroarch settings, i still get no sound... have tried all country settings.... this was the first thing i did. try everything at a stock level, no unibios and such.

**Comment by OBurgerGuy** [comment]
also the romset is valid and i dont use mame romsets on fba, like i said not new to this...

**Comment by OBurgerGuy** [comment]
rebuit bios with clrmamepro and YOUR .42 fba dat still need list of files may be missing some

**Comment by OBurgerGuy** [comment]
rebuilding romset and bios.... we shall see... we arent done yet boss

**Comment by OBurgerGuy** [comment]
also i am asking a personal favor of you to include a "FB Alpha v0.2.97.42 (ClrMame Pro XML, Neo-Geo only).dat.zip" thanks :)

**Comment by OBurgerGuy** [comment]
a "FB Alpha v0.2.97.42 (ClrMame Pro XML, Arcade only).dat.zip" would be awesome too!

**Comment by barbudreadmon** [comment]
So, do you intend to show me a video showing you are not running this with usa country set when you get this issue ? That's the only thing of interest to me : do you really have a sound issue when not using usa country. As i said, it is unrelated to mvs.

**Comment by OBurgerGuy** [comment]
![new-bios](https://user-images.githubusercontent.com/27814499/33593408-df24bc0a-d954-11e7-84ce-a08dd14be1ea.jpg)

**Comment by OBurgerGuy** [comment]
what settings do you wish to see as a control in video?

**Comment by OBurgerGuy** [comment]
are all my files accounted for in bios zip?

**Comment by barbudreadmon** [comment]
The unibios splashscreen when you launch a game show mode/country, i don't know why or how you hide it in your video.

**Comment by OBurgerGuy** [comment]
its not needed to see, looks unprofessional

**Comment by OBurgerGuy** [comment]
give me a few, will record and upload

**Comment by OBurgerGuy** [comment]
here ya go https://ufile.io/9cy0v

**Comment by barbudreadmon** [comment]
Ok, it seems real, and i'm unable to reproduce it, at this point i'm starting to think this is an issue between current fba and your custom unibios. Could you try with the default mvs bios which is "MVS Asia/Europe ver.6 (1 slot)" and tell me if you have sound ?

**Comment by OBurgerGuy** [comment]
funny thing is the custom unibios was left out, that was actually a pure rebuilt neogeo.zip with the free 3.2 and not my purchased custom 3.3 renamed to 3.2

**Comment by OBurgerGuy** [comment]
dude, do a lot of newbs just whine on here, you ha dme prove every fucking thing but my grandmothers maiden name, you have become jaded, are these issues that annoying? people post false shit that theyre too lazy to loo up? i mean ive had this issue for months, and have been researching it with no answer, finally i decided to float a ticket.

**Comment by OBurgerGuy** [comment]
you know i WAS going to give you the default in that video, but im like nahhh he wont want that

**Comment by OBurgerGuy** [comment]
ill record and post

**Comment by OBurgerGuy** [comment]
https://ufile.io/arwc1

**Comment by OBurgerGuy** [comment]
am i missing any files in the bios zip?

**Comment by thomasgoldstein** [comment]
OBurgerGuy, I can understand you're annoyed by barbudreadmon's attitude and all, but please don't rant or spam, and keep it straight to the point. For what it's worth, I decided to follow this repo and receive a mail whenever a new reply gets posted, I do not think there is any reason to post up to 6 comments in a row. This is very childish and makes the bug harder to follow. This isn't a chat. Thank you.

**Comment by barbudreadmon** [comment]
Yes, a lot of newbs whined here, especially those who didn't have a valid romset, that's why i added the warning about romsets on the "homepage".

No, your bios file seems fine now.

That's really weird, the mvs asia/europe bios (which is the default if you don't mess with core options) is what most people use (i never use anything else myself), yet you are the first one to report such issue, and i'm unable to reproduce it with the same settings in core options and/or unibios, except if i set country to usa.

Did you expect me to have no doubts about your issue happening whatever the mvs bios in those conditions ? While you gave me no information about your setup ? And after you mentioned "mvs usa" in your second post (which led me to find the issue with usa country, thanks for that) ? There are a lot of things you should have mentioned/done since the first post, checking your romset and mentioning you use a custom unibios are two of them, now let's complete the remaining ones : 
- What's your arch ? raspberry ? x86 ? x86_64 ?
- Could you post your retroarch.cfg and any override that could be included when launching this game ? Or at least tell me if rewind is enabled, and what's your audio setting.

**Comment by barbudreadmon** [comment]
I'm also interested in your retroarch version since i want to reproduce the exact same conditions.

**Comment by OBurgerGuy** [comment]
Can reply only, weather caused outage of internet, cannot send cfg currently, using cellular

**Comment by OBurgerGuy** [comment]
Ok Setup: raspberry pi 3 and retropie 4.3, only using lr-fba and select neogeo roms, the setup is included in a restored data east converted cabinet, not that that matters here. No rewind set, audio is default internally, setup in raspi-config to output to hdmi, this was a fresh 4.3 install not an upgrade from 4.2, I have 2 other setups a self contained arcade stick and a nes pi case, both raspberry pi 3 that I can try to see if they have this issue as well.

**Comment by OBurgerGuy** [comment]
I have never posted a problem on a git or thread like this so I am not familiar with the process or etiquette you’ll have to excuse my behavior as I am used to more of a forum-like format.  So that’s why you didn’t get a normal format of information and cause and effect.

**Comment by dankcushions** [comment]
You're still doing multiple comments. Every time you make a comment, 23 people get emailed. Please collect your thoughts into one comment.

Anyway. Retropie 4.3 won't use latest lr-fbalpha master or latest retroarch master, if that matters. You'd need to update lr-fblapha via source, but retroarch will be locked to the latest stable release at the time Retropie 4.3 was built.

**Comment by barbudreadmon** [comment]
Do you have a neogeo.zip file in your system folder ? If so, could you try copying the one in the fba roms folder to the system folder ? Someone pushed a commit 3 months ago to add system folder as a possible location for neogeo.zip, and i'm wondering if the neogeo.zip in system folder is not taking priority over the one in the roms folder.

**Comment by barbudreadmon** [comment]
Also, what's the gcc version in use for building this core in retropie ?

**Comment by barbudreadmon** [comment]
I removed the speedhack in neogeo driver, this speedhack seems to be known as a problem child, it fixed my issue with usa bios, let me know if it fixes your issue with mvs/unibios.

**Comment by OBurgerGuy** [comment]
Wait, so you removed the 100-200% overclock of the emulated cpu?

**Comment by barbudreadmon** [comment]
No, that's unrelated to the overclocking.

**Comment by OBurgerGuy** [comment]
Gucci version (Raspbian 4.9.2-10) 4.9.2

**Comment by OBurgerGuy** [comment]
Where am I to place bios? System folder? Please list exact directory

**Comment by barbudreadmon** [comment]
I don't know where is the system folder in retropie, if you don't know either i suppose you didn't put anything in it. So let's just try my last commit and tell me if it fixes your issue, which is likely since i had exactly the same sound issue limited to usa bios. Nevermind my other questions, that's just me being curious about why you got this issue with any mvs bios.

**Comment by OBurgerGuy** [comment]
yup, seems fixed nam1975 and baseball stars pro sound is fine unibios usa region

**Comment by OBurgerGuy** [comment]
on a side note may have broken roms now... looking into all 150 to see what plays and doesnt.

**Comment by OBurgerGuy** [comment]
sorry, supposed broken roms was a PEBKAC error, but i accidentally left one game out... with all the hubbub and bickering in the begining i forgot to add karnovr has the same sound issues as well during fighter selection screen ans in-game, the update from source did fix bstars and nam1975 but not karnovr, tested in all modes, again only us has issue. during testing all games for brokage i was reminded about karnovr...

**Comment by barbudreadmon** [comment]
Yes, i confirm i still have this issue with kanrovr, actually on this one i got it with any mvs bios too.

Btw, most of the code in the speedhack was actually targeted at s1945p, so if you feel like testing 1 game in depth, that's probably the one who needs it the most.

**Comment by OBurgerGuy** [comment]
Any word on karnovr?

**Comment by barbudreadmon** [comment]
http://neosource.1emulation.com/forums/index.php?topic=1731.840

So no, upstream remember fixing this issue at some point, but don't remember how, and don't know why and when the bug reappeared.

**Comment by OBurgerGuy** [comment]
The bug is not present in fba 2012, because that’s what I was using to remedy the problem.

**Comment by barbudreadmon** [comment]
After some testing, i think i finally understood the big issue here, including why you had the first issue with all mvs bios while i had it only for usa bios.

Fs files are at fault here, old version of fba generate fs files, and when those files are read by newer version, they potentially break things, removing "karnovr.fs" from retroarch's savefile folder fixed the issue with sounds in karnovr on my side, let me know if it does on yours (if you don't know where it is, check "savefile_directory" in your retroarch.cfg).

The fact "savefiles" are stored in the same directory for fba, fba2012, and probably the different mame flavors, with the same name, potentially overriding each other, is a big issue i think. The issue you had with all mvs bios probably came from having the games started with fba2012 at some point.

I think i'll have to move savefiles to a subdirectory, to be sure savefiles from another core are not interfering with this core. Having every arcade cores doing the same thing would probably be a good idea.

**Comment by OBurgerGuy** [comment]
.fs files are used for game and dipswitch settings, they typically get saved within the same directory as game-name.zip, it would be nice to know where the new bios location is, the roms directory is fine, but inhabe to dupe the zip for every directory, and traditionally it’s shown in the gamelist.xml but if you edit es_systems.cfg and remove the uppercase .ZIP then rename each of your bios to neogeo.ZIP fba will still honor the file, but ES will not show it in your game list.

**Comment by OBurgerGuy** [comment]
Noop! no good... deleted .fs file, set settings to what theyre supposed to be in Unibios, also it might benefit you to know i am using unibios 3.3 renamed to 3.2, stil curious why 3.3 hasnt been added, before you ask, yes... ive tested with 3.2 free version, and same results.

**Comment by barbudreadmon** [comment]
Could you give me a step by step tutorial of what you did when you say "deleted .fs file, set settings to what theyre supposed to be in Unibios" ? I'll try to reproduce, perhaps there is still another issue.

Anyway i'm 100% positive deleting karnovr.fs file fixed my issue at default fba settings, which means the loaded bios was asia-s3.rom, aka "MVS Asia/Europe ver.6 (1 slot)", and when i switched to unibios (i stayed with the default setting of Arcade/Europe) for testing, i still had music on select screen and ingame.

**Comment by OBurgerGuy** [comment]
It always worked on European settings, if you set it to USA arcade it won’t work. When you delete .fs file it sets unibios back to Europe default, by setting USA/arcade it creates. Brand new .fs file. As far as misunderstandings, I assumed just what you said, the only thing you have educate me on is that the force option overrides ANY bios setting below in the settings, whereas it will auto select an attributing rom, so here’s the thing then..  I don’t need a FULL bios, just USA and unibios 3.2 and the basic core files because I am only running USA arcade roms, and I’m running unibios not original bios.

**Comment by barbudreadmon** [comment]
Ok, so if i understand well because it's kinda unclear, after deleting karnovr.fs your issue only happens with unibios + arcade/usa settings ?
Normal mvs/aes bios and unibios with any setting besides arcade/usa work fine ?

**Comment by OBurgerGuy** [comment]
ok so yes... aaaand no.. yes deleting karnovr.fs and running any bios setting other than aes results in failure, so... unibios, force mvs, force dipswitch and choose bios, all produce no sound EXCEPT aes which appears to work fine, also have deleted the.fs file every time i tested with new bios settings and still no change.

**Comment by OBurgerGuy** [comment]
you explained what neo-epo.bin is what is neo-po.bin and do you have a complete list of bios files of what is what? i found this list but its incomplete:

sp-s2.sp1 (mainbios $0 - 128k) Europe MVS (Ver. 2)
sp-s.sp1 (mainbios $0 - 128k) Europe MVS (Ver. 1)
usa_2slt.bin (mainbios $0 - 128k) US MVS (Ver. 2?)
sp-e.sp1 (mainbios $0 - 128k) US MVS (Ver. 1)
asia-s3.rom (mainbios $0 - 128k) Asia MVS (Ver. 3)
vs-bios.rom (mainbios $0 - 128k) Japan MVS (Ver. 3)
sp-j2.sp1 (mainbios $0 - 128k) Japan MVS (Ver. 2)
sp1.jipan.1024 (mainbios $0 - 128k) Japan MVS (Ver. 1)
sp-45.sp1 (mainbios $0 - 128k) NEO-MVH MV1C
sp-1v1_3db8c.bin (mainbios $0 - 512k) Custom Japanese Hotel
uni-bios_2_3.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 2.3)
uni-bios_2_3o.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 2.3, older?)
uni-bios_2_2.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 2.2)
uni-bios_2_1.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 2.1)
uni-bios_2_0.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 2.0)
uni-bios_1_3.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 1.3)
uni-bios_1_2.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 1.2)
uni-bios_1_2o (mainbios $0 - 128k) Universe Bios (Hack, Ver. 1.2, older)
uni-bios_1_1.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 1.1)
uni-bios_1_0.rom (mainbios $0 - 128k) Universe Bios (Hack, Ver. 1.0)
sm1.sm1 (audiobios $0 - 128k)
sm1.sm1 (audi $0 - 128k)
000-lo.lo (zoomy $0 - 128k)
sfix.sfix (fixedbios $0 - 128k)
what is NEO-MVH MV1C...  usa, jap, eur?
the list is missing japan-j3.bin, neodebug.bin, neo-epo.bin, neo-po.bin, v2.bin and sp-u2.sp1

**Comment by barbudreadmon** [comment]
Ok, i was finally able to reproduce your issue by doing the following : 
- start the game (mvs or unibios arcade)
- insert coin so that the game go to the title screen
- wait for title screen music to start, immediately insert coin
- process further by pressing start, there is no music

Actually it does seem the issue only happens if you insert coin after the music start on title screen, i'll report this upstream.

**Comment by OBurgerGuy** [comment]
is there a way to get music by the actions you do after running the rom? like wait to insert coin?

**Comment by barbudreadmon** [comment]
Insert coins before title screen music begin, or insert coin after title screen disappear (select character screen and further).

**Comment by OBurgerGuy** [comment]
None of that resulted in sound...

**Comment by barbudreadmon** [comment]
I spent way too much time on this one, i did my last report about this upstream ( http://neosource.1emulation.com/forums/index.php?topic=1731.msg24806#msg24806 ), if they can't answer, i'll let it go.

**Comment by OBurgerGuy** [comment]
i can confirm your .fs file theory, as i was just playing Blazing Star (US/Arcade mode) to get my screenshot collection completed. and the first stage at the boss entry the music disappears, deleted the .fs file and unibios wanted its info again, set US/Arcade and Dont show splash, and got to boss again, there was sound this time, so yet another project to get to... delete all the old .fs files and remake.

**Comment by barbudreadmon** [comment]
I was suspecting this to be a potential source of issues for some time now, i'll move this core's nv/fs files to a subdirectory, it won't solve the issue with karnovr though since it happens with fresh karnovr.fs file.

I didn't withdraw from this issue yet, i still have a couple of things i wanna try, but i'm getting pessimistic about fixing it.

**Comment by OBurgerGuy** [comment]
Let me know what directory and where when you update the source

**Comment by barbudreadmon** [comment]
The core should now create a subdirectory called "fba" in the directory they used to be saved (it depends on your setup), nv/fs files will be loaded/saved there.

**Comment by OBurgerGuy** [comment]
ok.... more problems stemming probably from the updates to source "aof3" now gives me this error when running unibios usa/arcade, it never did before
![photo_2017-12-15_21-54-36](https://user-images.githubusercontent.com/27814499/34067127-a75f25d0-e1e2-11e7-9165-f8f3504a0506.jpg)

**Comment by OBurgerGuy** [comment]
tested the remaining 149, no issues, just this one, also the aof3 error screen doesnt appear in binary version FB Alpha v0.2.97.42 (retroarch 1.6.9), this issue seems specific to FB Alpha v0.2.97.42 26ee52f (updated from source)

**Comment by OBurgerGuy** [comment]
is this thread still being monitored? i would appreciate some feedback, barbudreadmon?

**Comment by barbudreadmon** [comment]
This is not a forum, don't bump topic. Yes this is monitored.

**Comment by OBurgerGuy** [comment]
sorry, thank you... wasn't sure i was heard. thanks

**Comment by OBurgerGuy** [comment]
soo... my other retropie setup that I updated to bab11d0 runs aof3 just fine in us mode and what not, but sometimes i have to set unibios settings again... like they disappear, but the .fs file is always there...?

**Comment by barbudreadmon** [comment]
Right now i'm trying to fix #162 , there is a buffer overflow and it could be the root of both issues, this buffer overflow could be in retroarch's core option system.

**Comment by OBurgerGuy** [comment]
i had a weird feeling it may be the high score dat, ill delete and report findings

**Comment by OBurgerGuy** [comment]
no difference, cant nail down what the issue is... will just wait for a fix to be sent upstream, please notify me when it happens.

**Comment by barbudreadmon** [comment]
The hiscore issue is just a symptom of the buffer overflow, we don't know yet if this is the hiscore code which is at fault for this overflow.

**Comment by barbudreadmon** [comment]
I can't reproduce the aof3 issue, do you still have it ?

**Comment by OBurgerGuy** [comment]
Yes

**Comment by OBurgerGuy** [comment]
Yes

**Comment by OBurgerGuy** [comment]
Accidentally closed

**Comment by barbudreadmon** [comment]
Even after updating to the latest commit and deleting your aof3.fs file ? I'm pretty sure a lot of issues with file corruption and settings not being saved properly were related to #162.

I tried MVS Europe/Asia, MVS Usa, and UNIBIOS 3.2 with USA/ARCADE settings. No problem here.

**Comment by OBurgerGuy** [comment]
I will test again on two separate set ups and get back to you possibly in the next couple days but definitely by Tuesday thanks again for your help with this issue

**Comment by barbudreadmon** [comment]
Actually i just tested karnovr with unibios, the sound works fine for me with any settings now. Only the genuine mvs bioses seems to have the sound issue.

**Comment by OBurgerGuy** [comment]
Nope, still no sound in karnovr and aof3 is still giving me the counterfeit board error

**Comment by barbudreadmon** [comment]
I'm wondering, how are you building this ? By retropie's scripts ? Or directly from command line ? And if you are using command line, what parameters are you using on the make command ?

**Comment by OBurgerGuy** [comment]
Retropie script “build from source”

**Comment by OBurgerGuy** [comment]
also, deleted hiscores.dat from "/opt/retropie/libretrocores/lr-fbalpha/metadata" and "/home/pi/RetroPie/BIOS/fba" to be sure i was compliant. no changes, what really gets me is the AOF3 counterfeit board error... why? and why now?

**Comment by barbudreadmon** [comment]
If i understand well, the issue with aof3 started for you after i moved the .fs files to a sub directory, which means you previously had an aof3.fs file which fixed this issue, probably generated by fbalpha2012. Did you try copying the aof3.fs file from fbalpha2012 to the fba subdirectory for this fbalpha ?

The .fs files are savestates, i have a feeling all of this could be somehow related to alignment issues (#166) in the savestates, that's why i asked how you built this, i don't use any "platform cflags" when i do my tests, i build this with a plain "make -f makefile.libretro". 

Actually, while fixing the issues in the savestates a few days ago, something came to mind : we are aligning parts of the savestates's code to 8 bytes adresses, which if i understand well is the size of a "normal" address for x86/x86_64, but if i'm not wrong, depending on the cflags, the compiler could end up using specific extensions of the cpu (sse2, avx, ...) which have a different address size, and probably can mess with the execution of the code in some odd way. I think arm or part of it (neon ?) could also have different address sizes.

I would be interested to know what @fr500 @twinaphex and/or whoever is more knowledgable than me in C/C++ (which is not difficult) think about this.

**Comment by OBurgerGuy** [comment]
I don’t have the original FS file for aof 3 on that system any longer but I think I have it on my supergun I tried putting everything back and it didn’t work but I don’t think it was the 2012 FS file let me try to put that back in the next couple hours from the supergun and we’ll see what happens

**Comment by barbudreadmon** [comment]
From the documentation, it seems arm uses 8 bytes addresses, but neon uses 16 bytes addresses, same as sse2, avx and the such. I'll try building on my raspberry with neon enabled to see if i can reproduce the same results than you.

**Comment by OBurgerGuy** [comment]
you were right about the math, used two aof3.fs files with different checksums and filesizes, thought they were from 2012 and current, but might have both been from current, found a backup of roms/fs files i made before upgrading from 2012 and boom! works, so heres the files see for yourself if you can:
[OburgerGuy-aof3-fs.zip](https://github.com/libretro/fbalpha/files/1589304/OburgerGuy-aof3-fs.zip)

**Comment by barbudreadmon** [comment]
Thanks, that's a great news, it basically means this issue is most likely savestates related. Now we just need to figure out in which way.

**Comment by barbudreadmon** [comment]
Sounds in karnovr seemed to work up to https://github.com/libretro/fbalpha/commit/228fb15c79382dba17f8ff4265a88b2425efe9d2, which is the commit prior to the .37 release. I'll compare .36 with .37 to see if i can find a reason for this.

**Comment by barbudreadmon** [comment]
Could you let me know if my latest commit fix the karnovr sound issue for you ? That's a dirty fix, but it works for me

**Comment by OBurgerGuy** [comment]
will do, updating from source now

**Comment by barbudreadmon** [comment]
Also, i was able to reproduce your issue with aof3 only once (started the game a few dozens of times, messed with various core options every time, at some point i got the same message about the board when i was trying to set unibios 3.2 through the dipswitches), and i don't know why. I deleted the aof3.fs file and was unable to reproduce it again after that. I won't be able to fix this one if i'm unable to reproduce it consistently.

**Comment by OBurgerGuy** [comment]
i think if you just address the math issue that you said was the root of the problem, the game will then create a proper .fs file, i think its due to the dev build making a corrupt .fs file that the aof3 issue exists.

**Comment by barbudreadmon** [comment]
The issue with savestates's size won't be totally fixed until the next release, and that's still only my theory : i should be able to reproduce it consistently by switching in and out some cflags, but it doesn't seem to work.

**Comment by OBurgerGuy** [comment]
ok... here it is. karnovr works perfectly even if the old fs file is deleted and a new one is set (usa/arcade) in unibios, sound for coin, selection screen and in-game seems all fine, how is this a dirty fix and why so uneasy about it? then aof3 obviously still the same, the if it creates an fs file from new settings, but if i insert the old fs file it works just fine, so this sint going to get fixed any time soon?

**Comment by OBurgerGuy** [comment]
what is the eta of the next release?

**Comment by barbudreadmon** [comment]
The issue with karnovr comes from the emulated cpu speed, i noticed the sound issue behaved differently depending on the overclock setting in core option, so i tried a few things, basically my "dirty fix" is forcing the cpu speed to 101% when it detects karnovr is launched, but that's defintely not a normal situation, especially since fba standalone works perfectly at the normal setting of 100.

For the "savestates fix", commiting it now would make bumping to next release a headache, since my usual method of "check if the change on the previous version is still necessary and apply it to the new one" would be enlarged by changes on around 50 files.

I don't know when is the next release.

**Comment by OBurgerGuy** [comment]
i have my neogeo cpu overclock set to 190 or 200, no audio, setting it to 100, 110, 120, 130, 140, 150, 160, 170, 180 i get audio back this is .42 bab11d0, so you can remove "dirty fix" and alert people not to OC past 180. the aof3 may be tied to this as well, come to think about it the fact that it may be writing corrupt fs files may be part of the 180-200 CPU OC too.

**Comment by OBurgerGuy** [comment]
YUP just tested the theory, aof3 shows error when you OC CPU to 190 or 200, it creates a bad fs file too, if you delete fs file and set to OC 180 everythings fine!

**Comment by OBurgerGuy** [comment]
sorry about the edits, wanted to be accurate

**Comment by barbudreadmon** [comment]
Do you mean you were always running the games at OC 200 ?

**Comment by OBurgerGuy** [comment]
yes, i thought i mentioned that before... or atleast i am under that impression.

**Comment by barbudreadmon** [comment]
That explains a lot about the issues i couldn't reproduce myself. It was probably in your video and i overlooked this. Anyway it allowed me to notice the actual sound code for neogeo (there were quite a few fixes in the last year due to issues with s1945p) is extremely picky with this OC value.

I'm not sure this option is still available in fba standalone, actually while working on this issue i have been thinking about removing 2 core options from this core, and it is one of them (the other is the neogeo mode : it is a relic from when we had no access to the bios dipswitch, and it is kinda confusing and unsafe since we now have it). I'm not even sure it still affects any game in a positive way (i remember people mentioning it helped with games having slowdowns on the original hardware, but is that still true ?), i don't remember using it once myself.

**Comment by OBurgerGuy** [comment]
the slowdowns are gone in games if you OC 150 or higher, 180-200 is sweet!

**Comment by barbudreadmon** [comment]
Do you know if there is a list of those neogeo games affected by those slowdowns ? I remember twinkle star sprites being one of them, i kinda remember there was one of the mslug too (but i think this one possess a hacked version which shouldn't have those slowdowns), i don't know about the others.

**Comment by OBurgerGuy** [comment]
I know from personal experience that all of the metal slug games have the issue, It has slow down when there’s more than a few sprites on the screen at once like when you get a lot of enemies I know that happens constantly and metal slug three or four

**Comment by AkimanBengus** [comment]
> (the other is the neogeo mode : it is a relic from when we had no access to the bios dipswitch, and it is kinda confusing and unsafe since we now have it)

Hello @barbudreadmon, I know the core option "NeoGeo Mode" can be confusing but when you understand it, not specially ;-)

When I implemented the "DipSwitches" options I thought like you but now I think it would be good to keep it.
For exemple I use it to have a menu, in EmulationStation, dedicated only to AES to play all games in Console Mode and another menu to have the MVS ones to play in Arcade Mode.

If we remove the core option "NeoGeo Mode", this would be still possible but I will have to change it for every games because "DipSwitches" are game per game.

Best regards

**Comment by barbudreadmon** [comment]
Hello @AkimanBengus , long time no see :).

Yes that's why i've kept it until now, even though i had to fix it again not so long ago for a special case (see irrmaze in #138).

Since i understand it, i think that's a good idea, but it's confusing for a lot of people (see the few videos in this issue, @OBurgerGuy was setting both the mode core option and the bios dipswitch), so it doesn't help with debug.

**Comment by AkimanBengus** [comment]
Yes long time. I haven't much time to program and play like before :-(

I saw some issues that came from the misunderstood of people and I wanted to help to explain but I didn't take time for it because I wanted to test before...

It was the best I could do with the core option system of "Retroarch"...
It would be much better if the system was more flexible, like to have the possibility to indicate a Default value for exemple. I had other ideas 2 years ago but now I don't remember :-)

Maybe rename it will help people to understand, even if this is clear enough for me ;-)

**Comment by barbudreadmon** [comment]
> Maybe rename it will help people to understand, even if this is clear enough for me ;-)

Yes, that's what i did, now it is named "prefered mode if available" or something like this, doesn't seem to be enough though.

**Comment by barbudreadmon** [comment]
The "aof3 at OC 200" issue won't be fixed, looks to me like the game itself is detecting an abnormal clock on the cpu, hence reporting about the board not being an official mvs board, we can't do anything about that.

The last commit should fix all issues with sound on neogeo properly, can i close ?

**Comment by OBurgerGuy** [comment]
Did you remove dirty fix of karnovr? As this new information makes it invalid

**Comment by barbudreadmon** [comment]
Yes i did. I don't know why the build for wiiu failed, you shouldn't care about that, travis failed on half of the commit since it was deployed on this repo, even though 100% of those commit were actually building successfully

**Comment by OBurgerGuy** [comment]
Ok, thanks... will keep an eye on build, waiting for final release. You can close, thanks for being the awesome attention to detail and the help, will open a new issue if another arises.

**Comment by dinkc64** [comment]
yay! finally fixed :)
